### PR TITLE
update kubernetes dependencies to v1.25.0

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

> /kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
This release update kubernetes module dependencies to v1.25 
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -16,13 +16,13 @@ require (
...
-	k8s.io/component-helpers v0.24.3
-	k8s.io/klog/v2 v2.60.1
+	google.golang.org/grpc v1.47.0
+	k8s.io/api v0.25.0
+	k8s.io/apimachinery v0.25.0
+	k8s.io/client-go v0.25.0
+	k8s.io/component-base v0.25.0
+	k8s.io/component-helpers v0.25.0
+	k8s.io/klog/v2 v2.70.1
 	k8s.io/kubernetes v1.24.3
```
Comment (xing-yang) [inline go.mod:26]: Any reason why this one is not updated to 1.25?

File: go.mod [hunk 2]
```diff
@@ -54,53 +54,53 @@ require (
...
+	k8s.io/apiserver => k8s.io/apiserver v0.25.0
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.25.0
+	k8s.io/client-go => k8s.io/client-go v0.25.0
+	k8s.io/cloud-provider => k8s.io/cloud-provider v0.25.0
+	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.25.0
+	k8s.io/code-generator => k8s.io/code-generator v0.25.1-rc.0
+	k8s.io/component-base => k8s.io/component-base v0.25.0
+	k8s.io/component-helpers => k8s.io/component-helpers v0.25.0
+	k8s.io/controller-manager => k8s.io/controller-manager v0.25.0
+	k8s.io/cri-api => k8s.io/cri-api v0.25.1-rc.0
```
Comment (xing-yang) [inline go.mod:91]: Why rc.0?

File: go.mod [hunk 3]
```diff
@@ -54,53 +54,53 @@ require (
...
+	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.25.0
+	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.25.0
+	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.25.0
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.25.0
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.25.0
+	k8s.io/kubectl => k8s.io/kubectl v0.25.0
+	k8s.io/kubelet => k8s.io/kubelet v0.25.0
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.25.0
+	k8s.io/metrics => k8s.io/metrics v0.25.0
+	k8s.io/mount-utils => k8s.io/mount-utils v0.25.1-rc.0
```
Comment (xing-yang) [inline go.mod:101]: Why rc.0?

File: go.mod (hunk 1 above)
Comment (humblec) [reply go.mod]: for some reason it was not picked up.. have corrected now.

File: go.mod (hunk 2 above)
Comment (humblec) [reply go.mod]: corrected it. @xing-yang

File: go.mod (hunk 3 above)
Comment (humblec) [reply go.mod]: corrected.

**Comment by humblec** [comment]
@xing-yang I have addressed review comments. ptal.. thanks

**Pull request merged**